### PR TITLE
Exception message added for Virtual or Empty Cart

### PR DESCRIPTION
**Pull request by qualwebs**
Added exception message for Virtual Products or Empty Cart 

### Description
If Virtual products are added in cart, Shipping methods are no applicable which is fine. But it was returning blank array []. Instead of blank it should show a exception message which I have added.

Message is always helpful for developers to identify error/exception. Specially with REST APIs.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios
1. Add a virtual product in cart.
2. Estimate shipping
3. It was returning blank array [], I just added a exception message instead of blank array.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=15562) <br/>All committers have signed the CLA.

**Review by larsroettig — CHANGES_REQUESTED** [review]
We can not merge this it is not backwards compatible:

https://devdocs.magento.com/guides/v2.2/extension-dev-guide/versioning/codebase-changes.html

You changed - the format of the returned method result. 

With this change, you also break two unit tests.

Pls, open an Issue if this accepts we can use this fix for 2.3.

**Comment by larsroettig** [comment]
Hi @qualwebs, I close this issue because I discussed this change with other Maintainer. 
In our opinion, when there are zero payment methods available, an empty array is fine.

best regards,

Lars

---
Maintainer at Magento Community Engineering

**Pull request closed without merging**
